contracts/OctobayGovNFT.sol

```diff
@@ -79,18 +95,23 @@ contract OctobayGovNFT is OctobayStorage, ERC721Pausable {
         _unpause();
         _safeMint(_to, tokenId);
         _pause();
-        tokenIDsByGovToken[_govToken] = tokenId;
+        govTokensByTokenId[tokenId] = _govToken;
         emit MintNFTForGovTokenEvent(_to, tokenId, address(_govToken));
         return tokenId;
     }
 
+    /// TODO: I just realised a bug with this, it's possible for a user to have multiple NFTs per gov token here.
+    /// An attacker could actually mint and send an NFT with no permissions and send it to someone to 'cover' or 'hide'
+    /// their actual NFT. I suppose instead we should have a function which checks whether a user simply has a specific permission
+    /// for a given gov token and look through all of their list of NFTs.
+
```

Comment:
First.... :kissing_heart: for always thinking about these cases.

> and send it to someone to 'cover' or 'hide' their actual NFT.

What about simply not allowing this? Only one NFT per user and gov token.

Reply:
Hmmm, yeah I think ensuring each address only has one NFT is probably a simpler way of handling things.

But then we can't ignore an incoming transfer to someone that already has one right? We essentially need to OR the array of permissions in their existing NFT with the incoming one right?

Reply:
And then we burn the original NFT that was transferred to them? I feel like this needs more consideration...

Reply:
I just realised if we do use this, we also need to think about minting as well, if someone mints to someone with an existing NFT for that gov token we'll need to do the same thing.

Reply:
Don't we just need a check in the transfer function of the NFT contract?

When minting new tokens for an address that already has one for that gov token, the frontend can inform the user that this is not possible and the transaction will fail. Transfers will also only happen through our frontend. I was actually doubting this whole NFT idea for a moment yesterday. Could all be implemented without ERC721. But on the other hand I thought then.... well... we just use that standard as a starting point. Totally fine.

Reply:
Yeah it's not like ERC721 is making our lives harder at all, we do still get some stuff for free. Plus we get to ride on the back of all the NFT memes :D

I can see cases where minting to someone that has a token makes sense though. Someone joins the org and you give them limited permissions, then later on you want to upgrade their permissions. Minting would be the simplest way of doing this at the moment, unless we have a way for owners to grant new permissions on NFTs.

Reply:
I would have to think about the actual implementation but maybe it is then better to just allow multiple tokens and iterate over them to find a needed permission. Why should we restrict normal functionality? This will always be related to a certain gov token. Noone will have like 100 NFTs for a governance token. And since the Subgraph is indexing everything anyway, the lookups will happen there and everytime an on-chain check is needed, we provide the id of the suitable token, found through the Subgraph and on-chain its just a simple owner and permission check. Does that make sense?

Reply:
I think the idea of having only one NFT and OR'ing the permissions together is fine still. When someone mints it really is a transfer more or less, so we can treat it the same.

Reply:
For the record, as discussed in discord: We are going with the approach you suggested in the first place.

> I suppose instead we should have a function which checks whether a user simply has a specific permission for a given gov token and look through all of their list of NFTs.

This preserves default NFT functionality, that might or might not be needed in the future but for now we are in experimenting mode anyway.